### DISCUSSION TITLE
Include file with variable name

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#4085] [General]
Hi everyone, is posible to include files with a variable name? I tried like the example below but it didn't work:

```
if file.exists("/home/extra-#{account}.conf") then
%include "/home/extra-#{account}.conf"
end
```

The error I got:

`File "/home/functions.conf", line 96, char 66: cannot %include, file "/home/extra-#{account}.conf" doesn't exist.`

**Comment by toots** [comment]
This is unfortunately not available. File inclusion is done while parsing the script so the variables are not yet interpreted/known.

You could potentially use `eval` but you would have to return something in your file:

Included file:
```liquidsoap
{ password = "super secret",
  api_key = "super secret" }
```

main file:
```liquidsoap
let eval (config: {password: string, api_key: string}) = file.contents("/home/extra-#{account}.conf")
```
